bazel_defaults

```diff
@@ -66,23 +66,22 @@ build:gdr --define=with_gdr_support=true
 build:ngraph --define=with_ngraph_support=true
 build:verbs --define=with_verbs_support=true
 
+
 build --define=use_fast_cpp_protos=true
 build --define=allow_oversize_protos=true
 build --define=grpc_no_ares=true
 
-build --spawn_strategy=standalone
-build --genrule_strategy=standalone
-build -c opt
+
+#build --spawn_strategy=standalone
```

Comment:
@samikama why commenting out these?

Reply:
having -c opt forces bazel to do opt builds preventing users from having debug builds. Users need to edit .bazelrc to be able to debug. AFAIK default of bazel is fastbuild which is pretty close to opt. --spawn_strategy default is sandboxed where it can and standalone if sandboxing doesn't work. is there a strong reasond against sandboxing? AFAIK this was default before this file was added.

Reply:
Fastbuild is not close to opt, it disables most optimizations (builds fast, runs slow).

Reply:
You can always to --copt=-g to add the debug symbols.
This does not prevent you from having debug builds.

Reply:
@martinwicke, you are right, standing corrected. But what is going to happen if user wants to do a debug build? Isn't it better to specify this flag from the command-line?

Reply:
@gunan that is still -O2 -g, not really debug i.e. -O0 -g

Reply:
By the way, with no extra parameters to bazel command-line  and without "build -c opt" flag in bazelrc, TF seems to be building with -g0 -O2 -DNDEBUG .

Reply:
I just checked out and experimented with the two following commands:
```
1 - bazel build -s tensorflow/tools/pip_package:build_pip_package
2 - bazel build -s -c dbg tensorflow/tools/pip_package:build_pip_package
```
`-s` forces bazel to write out the subcommands.
Here is what I see. With the first command, I verified that `-O2` is passed to my command:
```
/usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF
```
However, When I run the second command, I see no O2 passed to the command:
```
/usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g '-std=c++0x' -MD -MF 
```

So even though we have this in our bazelrc, the option I provide through the command line overrides the option in bazelrc. Therefore, I do not think having `-c opt` here forces users out of a debug build.

Reply:
@gunan, can you try with removing the flag from .bazelrc? in my case it seems to default to opt. Anyway, I can add it back.

Reply:
When I remove `build -c opt` line from my bazelrc, this is what I see in subcommands:
```
/usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF ...
```
So I do not think it defaults to -c opt`

Reply:
Is it possible that you have a bazelrc at your home folder, or a system level bazelrc file that changes your defaults?

Reply:
I am pretty sure that is not the case but started from scratch again. Regardless, I added the flag back.